core/ledger/kvledger/txmgmt/txmgr/tx_simulator.go

```diff
@@ -157,7 +179,15 @@ func (s *txSimulator) GetTxSimulationResults() (*ledger.TxSimulationResults, err
 		return nil, s.queryExecutor.err
 	}
 	s.queryExecutor.addRangeQueryInfo()
-	return s.rwsetBuilder.GetTxSimulationResults()
+	simResults, err := s.rwsetBuilder.GetTxSimulationResults()
+	if err != nil {
+		return nil, err
+	}
+	simResults.PrivateReads = s.privateReads.Clone()
+	for _, policy := range s.keySignaturePolicies {
+		simResults.KeySignaturePolicies = append(simResults.KeySignaturePolicies, policy)
+	}
```

Comment:
Is this cloning of both required?

Reply:
Yes, I've put a comment in the code explaining.  There is an existing unit test that catches this.